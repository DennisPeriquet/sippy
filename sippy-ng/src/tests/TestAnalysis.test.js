/** @jest-environment setup-polly-jest/jest-environment-node */

import 'jsdom-global/register'
import { act, waitFor } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import {
  expectLoadingPage,
  setupDefaultPolly,
  withoutMuiID,
} from '../setupTests'
import { mount } from 'enzyme'
import { QueryParamProvider } from 'use-query-params'
import { TestAnalysis } from './TestAnalysis'
import React from 'react'

jest.useRealTimers()

describe('TestAnalysis', () => {
  setupDefaultPolly()

  it('should render correctly', async () => {
    const fetchSpy = jest.spyOn(global, 'fetch')

    let wrapper
    await act(async () => {
      wrapper = mount(
        <QueryParamProvider>
          <BrowserRouter>
            <TestAnalysis
              release="4.8"
              test="[sig-api-machinery] Kubernetes APIs remain available with reused connections"
            />
          </BrowserRouter>
        </QueryParamProvider>
      )
    })

    expectLoadingPage(wrapper).toBeTruthy()

    await waitFor(() => {
      wrapper.update()
      expectLoadingPage(wrapper).toBeFalsy()
    })

    expect(wrapper.text()).toContain('Pass rate')
    expect(wrapper.exists()).toBe(true)
    expect(withoutMuiID(wrapper)).toMatchSnapshot()
    expect(fetchSpy).toHaveBeenCalledTimes(3)
  })
})
