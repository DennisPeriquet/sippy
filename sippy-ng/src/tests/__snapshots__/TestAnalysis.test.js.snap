// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestAnalysis should render correctly 1`] = `
<nav class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
     aria-label="breadcrumb"
>
  <ol class="MuiBreadcrumbs-ol">
    <li class="MuiBreadcrumbs-li">
      <a href="/release/4.8">
        Overview
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <a href="/tests/4.8">
        Tests
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <p class="MuiTypography-root MuiTypography-body1">
        Test Analysis
      </p>
    </li>
  </ol>
</nav>
<div class="MuiContainer-root MuiContainer-maxWidthLg">
  <h3 class="MuiTypography-root MuiTypography-h3"
      style="text-align: center;"
  >
    Test Analysis
  </h3>
  <hr class="MuiDivider-root"
      style="margin: 20px;"
  >
  <div class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3">
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4">
      <div class="MuiPaper-root MuiCard-root makeStyles-summaryCard-2 makeStyles-summaryCard-3 MuiPaper-elevation5 MuiPaper-rounded"
           style="background-color: rgb(152, 196, 122);"
           title
      >
        <div class="MuiCardContent-root makeStyles-cardContent-1">
          <h6 class="MuiTypography-root MuiTypography-h6">
            Overall
          </h6>
          <div align="center">
            <div style="width: 70%;">
            </div>
            <span title="263 runs"
                  class
            >
              99.50%
            </span>
            <svg class="MuiSvgIcon-root"
                 focusable="false"
                 viewbox="0 0 24 24"
                 aria-hidden="true"
                 data-icon="ArrowUpwardRoundedIcon"
                 style="stroke: green; stroke-width: 3; color: green;"
            >
              <path d="M13 19V7.83l4.88 4.88c.39.39 1.03.39 1.42 0 .39-.39.39-1.02 0-1.41l-6.59-6.59a.9959.9959 0 00-1.41 0l-6.6 6.58c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L11 7.83V19c0 .55.45 1 1 1s1-.45 1-1z">
              </path>
            </svg>
            <span title="348 runs"
                  class
            >
              94.51%
            </span>
          </div>
        </div>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-8">
      <div class="MuiPaper-root MuiCard-root test-failure-card MuiPaper-elevation5 MuiPaper-rounded"
           style="height: 100%;"
      >
        <h5 class="MuiTypography-root MuiTypography-h5">
          [sig-api-machinery] Kubernetes APIs remain available with reused connections
        </h5>
        <hr class="MuiDivider-root test-divider">
        <div align="center">
          <a class="MuiButtonBase-root MuiButton-root MuiButton-contained test-button MuiButton-containedPrimary"
             tabindex="0"
             aria-disabled="false"
             target="_blank"
             href="https://bugzilla.redhat.com/enter_bug.cgi?classification=Red%20Hat&amp;product=OpenShift%20Container%20Platform&amp;cf_internal_whiteboard=buildcop&amp;short_desc=%5Bsig-api-machinery%5D%20Kubernetes%20APIs%20remain%20available%20with%20reused%20connections&amp;comment=%0A%5Bsig-api-machinery%5D%20Kubernetes%20APIs%20remain%20available%20with%20reused%20connections%0A%0Ais%20failing%20frequently%20in%20CI%2C%20see%3A%0Ahttps%3A%2F%2Fsippy.ci.openshift.org%2Fsippy-ng%2Ftests%2F4.8%2Fanalysis%3Ftest%3D%255Bsig-api-machinery%255D%2520Kubernetes%2520APIs%2520remain%2520available%2520with%2520reused%2520connections%0A%0AFIXME%3A%20Replace%20this%20paragraph%20with%20a%20particular%20job%20URI%20from%20the%20search%20results%20to%20ground%20discussion.%20%20A%20given%20test%20may%20fail%20for%20several%20reasons%2C%20and%20this%20bug%20should%20be%20scoped%20to%20one%20of%20those%20reasons.%20%20Ideally%20you'd%20pick%20a%20job%20showing%20the%20most-common%20reason%2C%20but%20since%20that's%20hard%20to%20determine%2C%20you%20may%20also%20chose%20to%20pick%20a%20job%20at%20random.%20%20Release-gating%20jobs%20(release-openshift-...)%20should%20be%20preferred%20over%20presubmits%20(pull-ci-...)%20because%20they%20are%20closer%20to%20the%20released%20product%20and%20less%20likely%20to%20have%20in-flight%20code%20changes%20that%20complicate%20analysis.%0A%0AFIXME%3A%20Provide%20a%20snippet%20of%20the%20test%20failure%20or%20error%20from%20the%20job%20log%0A&amp;version=4.9&amp;cc=sippy@dptools.openshift.org"
          >
            <span class="MuiButton-label">
              Open a new bug
            </span>
            <span class="MuiTouchRipple-root">
            </span>
          </a>
          <a class="MuiButtonBase-root MuiButton-root MuiButton-contained test-button MuiButton-containedSecondary"
             tabindex="0"
             aria-disabled="false"
             target="_blank"
             href="https://search.ci.openshift.org/?search=%5C%5Bsig-api-machinery%5C%5D%20Kubernetes%20APIs%20remain%20available%20with%20reused%20connections&amp;maxAge=336h&amp;context=1&amp;type=bug%2Bjunit&amp;name=&amp;excludeName=&amp;maxMatches=5&amp;maxBytes=20971520&amp;groupBy=job"
          >
            <span class="MuiButton-label">
              Search CI Logs
            </span>
            <span class="MuiTouchRipple-root">
            </span>
          </a>
          <a class="MuiButtonBase-root MuiButton-root MuiButton-contained test-button"
             tabindex="0"
             role="button"
             aria-disabled="false"
             href="/jobs/4.8/runs?filters=%7B%22items%22%3A%5B%7B%22id%22%3A99%2C%22columnField%22%3A%22failedTestNames%22%2C%22operatorValue%22%3A%22contains%22%2C%22value%22%3A%22%5Bsig-api-machinery%5D%20Kubernetes%20APIs%20remain%20available%20with%20reused%20connections%22%7D%5D%7D&amp;sortField=timestamp&amp;sort=desc"
          >
            <span class="MuiButton-label">
              See all job runs
            </span>
            <span class="MuiTouchRipple-root">
            </span>
          </a>
        </div>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12">
      <div class="MuiPaper-root MuiCard-root test-failure-card MuiPaper-elevation5 MuiPaper-rounded">
        <h5 class="MuiTypography-root MuiTypography-h5">
          Pass rate
          <svg class="MuiSvgIcon-root"
               focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               title="Test pass rate is approximated as how many job runs on the given day without a test failure. Only jobs with at least one failure over the reporting period are shown individually."
          >
            <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z">
            </path>
          </svg>
        </h5>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4">
      <div class="MuiPaper-root MuiCard-root test-failure-card MuiPaper-elevation5 MuiPaper-rounded"
           style="height: 100%;"
      >
        <h5 class="MuiTypography-root MuiTypography-h5">
          Failure count by variant
        </h5>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-8">
      <div class="MuiPaper-root MuiCard-root test-failure-card MuiPaper-elevation5 MuiPaper-rounded"
           style="height: 100%;"
      >
        <a class="MuiTypography-root MuiTypography-h5"
           style="margin-bottom: 20px;"
           href="/jobs/4.8/runs?filters=%7B%22items%22%3A%5B%7B%22id%22%3A99%2C%22columnField%22%3A%22failedTestNames%22%2C%22operatorValue%22%3A%22contains%22%2C%22value%22%3A%22%5Bsig-api-machinery%5D%20Kubernetes%20APIs%20remain%20available%20with%20reused%20connections%22%7D%5D%7D&amp;sortField=timestamp&amp;sort=desc"
        >
          Job runs with test failure
        </a>
        <div class="MuiBackdrop-root"
             aria-hidden="true"
             style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          Fetching data...
          <div class="MuiCircularProgress-root MuiCircularProgress-indeterminate"
               style="width: 40px; height: 40px;"
               role="progressbar"
          >
            <svg class="MuiCircularProgress-svg"
                 viewbox="22 22 44 44"
            >
              <circle class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                      cx="44"
                      cy="44"
                      r="20.2"
                      fill="none"
                      stroke-width="3.6"
              >
              </circle>
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12">
      <div class="MuiPaper-root MuiCard-root test-failure-card MuiPaper-elevation5 MuiPaper-rounded"
           style="height: 100%;"
      >
        <p class="MuiTypography-root MuiTypography-body1">
          Linked bugs
          <svg class="MuiSvgIcon-root"
               focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               title="Linked bugs are bugs that mention the failing test name
and are targeted to the release being reported on."
          >
            <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z">
            </path>
          </svg>
        </p>
        <div class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
             style="margin-top: 20px;"
        >
          <table class="MuiTable-root"
                 aria-label="bug-table"
          >
            <thead class="MuiTableHead-root">
              <tr class="MuiTableRow-root MuiTableRow-head">
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Bug ID
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Summary
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Status
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Component
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Target Release
                </th>
              </tr>
            </thead>
            <tbody class="MuiTableBody-root">
              <tr class="MuiTableRow-root">
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    scope="row"
                >
                  <a href="https://bugzilla.redhat.com/show_bug.cgi?id=1921157">
                    1921157
                  </a>
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  [sig-api-machinery] Kubernetes APIs remain available for new connections
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  ASSIGNED
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  Node
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  4.8.0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12">
      <div class="MuiPaper-root MuiCard-root test-failure-card MuiPaper-elevation5 MuiPaper-rounded"
           style="height: 100%;"
      >
        <p class="MuiTypography-root MuiTypography-body1">
          Associated bugs
          <svg class="MuiSvgIcon-root"
               focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               title="Associated bugs are bugs that mention the failing test
name but are not targeted to the release being reported on."
          >
            <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z">
            </path>
          </svg>
        </p>
        <div class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
             style="margin-top: 20px;"
        >
          <table class="MuiTable-root"
                 aria-label="bug-table"
          >
            <thead class="MuiTableHead-root">
              <tr class="MuiTableRow-root MuiTableRow-head">
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Bug ID
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Summary
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Status
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Component
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Target Release
                </th>
              </tr>
            </thead>
            <tbody class="MuiTableBody-root">
              <tr class="MuiTableRow-root">
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    scope="row"
                >
                  <a href="https://bugzilla.redhat.com/show_bug.cgi?id=1983829">
                    1983829
                  </a>
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  ovn-kubernetes upgrade jobs are failing disruptive tests
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  ASSIGNED
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  Networking
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  4.9.0
                </td>
              </tr>
              <tr class="MuiTableRow-root">
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    scope="row"
                >
                  <a href="https://bugzilla.redhat.com/show_bug.cgi?id=1943334">
                    1943334
                  </a>
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  [ovnkube] node pod should taint NoSchedule on termination; clear on startup
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  VERIFIED
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  Networking
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  4.9.0
                </td>
              </tr>
              <tr class="MuiTableRow-root">
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    scope="row"
                >
                  <a href="https://bugzilla.redhat.com/show_bug.cgi?id=1983758">
                    1983758
                  </a>
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  upgrades are failing on disruptive tests
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  NEW
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  Routing
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  ---
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
